### PR TITLE
[ci] [appveyor] Move Appveyor to OPAM 2.

### PR DESCRIPTION
**Pull request by ejgallego**
We update the Appveyor configuration so it uses OPAM 2.0, and thus it
can install newer packages.

### CODE REVIEW

**Comment by SkySkimmer** [comment]
Looks like it hung somehow. Doesn't look spurious.

**Comment by ejgallego** [comment]
It turns out the problem is the cache; I don't have permissions in appveyor thus I'll have to workaround this.

File: dev/ci/appveyor.sh [hunk 1]
```diff
@@ -2,14 +2,15 @@
...
+wget https://github.com/fdopen/opam-repository-mingw/releases/download/0.0.0.2/opam64.tar.xz -O opam64.tar.xz
 tar -xf opam64.tar.xz
 bash opam64/install.sh
 
-opam init -a mingw https://github.com/fdopen/opam-repository-mingw.git --comp $APPVEYOR_OPAM_SWITCH --switch $APPVEYOR_OPAM_SWITCH
-eval "$(opam config env)"
-opam install -y num ocamlfind ounit
+opam init default -a -y "https://github.com/fdopen/opam-repository-mingw.git#opam2" -c $APPVEYOR_OPAM_VARIANT --disable-sandboxing
+eval "$(opam env)"
+opam install -y num ocamlfind ounit dune
```
Comment (SkySkimmer) [inline dev/ci/appveyor.sh]: Why install dune here?

**Comment by SkySkimmer** [comment]
Looks like it works now.

File: dev/ci/appveyor.sh (hunk 1 above)
Comment (ejgallego) [reply dev/ci/appveyor.sh]: Oh that's a leftover of #8421 , from which this PR was spun.

**Comment by ejgallego** [comment]
Let's see what happens with the cache again.

**Comment by ejgallego** [comment]
So this is ready for merge as I works if the cache is not used; I think the best strategy is merge, then clean the caches.

**Comment by ejgallego** [comment]
Working example [cache disabled]https://ci.appveyor.com/project/coq/coq/builds/20516158

**Comment by maximedenes** [comment]
Oh! Am I still listed as an appveyor maintainer? Anyway, will take care of this one.

How should the appveyor be read?

**Comment by ejgallego** [comment]
> Oh! Am I still listed as an appveyor maintainer? Anyway, will take care of this one.

Yes you are; feel free to move to me [tho this wouldn't help a lot as I am the one mainly caring] or to `ci-maintainers if @SkySkimmer agrees.

Note that a problem we have is that we are not admins in the appveyor instance so you should give us access.

> How should the appveyor be read?

What do you mean?

For this PR, there seems to be a problem with the cache that I cannot track, so you need to merge, then clean the caches and have this run I think.

**Comment by maximedenes** [comment]
> For this PR, there seems to be a problem with the cache that I cannot track, so you need to merge, then clean the caches and have this run I think.

Ok, so you suggest I merge even if it failed? I'm ok with it, just asking to make sure I don't do a mistake.

**Comment by ejgallego** [comment]
> Ok, so you suggest I merge even if it failed? I'm ok with it, just asking to make sure I don't do a mistake.

If you look a bit up you can see a build without cache that suceeded. I can't make this one to succeed as I don't have the proper permissions to clean the cache in appveyor.

So you need to merge, clean the caches, then rerun I think.

**Review by maximedenes — APPROVED** [review]

**Comment by SkySkimmer** [comment]
This seems risky, unrebased branches still trigger appveyor so there will be cache resurrections for a while.
Shouldn't we make the `  - C:\cygwin64 -> dev\ci\appveyor.bat` cache depend on the appveyor.sh too?

**Comment by ejgallego** [comment]
We already have:
```
  - C:\cygwin64\home\appveyor\.opam -> dev\ci\appveyor.sh
```
why is that not working?

**Comment by SkySkimmer** [comment]
What's bash profile doing?

Gaëtan Gilbert

**Comment by ejgallego** [comment]
This seems to be working now, failed due to overtime, rerunning again but if the worker is fast it should go thru.

**Comment by Zimmi48** [comment]
Passed.

**Comment by ejgallego** [comment]
Thanks!

**Pull request merged**